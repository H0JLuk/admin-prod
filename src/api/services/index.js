import { getSession } from './sessionService';

export const getReqOptions = () => {
    const token = getSession();
    return {
        headers: { token }
    };
};
