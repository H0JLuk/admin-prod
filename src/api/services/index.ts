import { ApiRequestHeaders, ApiRequestOptions } from '@types';
import { getDefaultAppCode } from './clientAppService';
import { getAppCode, getSession } from './sessionService';

const TOKEN_HEADER = 'token';
const APP_CODE_HEADER = 'clientAppCode';
const CONTENT_TYPE = 'Content-Type';
const ACCEPT = 'Accept';

export const getReqOptions = (contentType?: string, currAppCode?: string): ApiRequestOptions => {
    const token = getSession();
    const appCode = currAppCode || getAppCode() || getDefaultAppCode(); // TODO: remove getDefaultAppCode after second phase
    const headers = {} as ApiRequestHeaders;
    token && (headers[TOKEN_HEADER] = token);
    appCode && (headers[APP_CODE_HEADER] = appCode);
    contentType && (headers[ACCEPT] = '*/*');
    contentType && (headers[CONTENT_TYPE] = contentType);
    return { headers };
};
