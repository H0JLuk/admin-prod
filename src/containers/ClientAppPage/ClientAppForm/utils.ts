import { customNotifications } from '@utils/notifications';
import { ArgsProps } from 'antd/lib/notification';
import { addSettings, updateSettingsList } from '@apiServices/settingsService';
import { SETTINGS_TYPES } from './ClientAppFormConstants';
import { SettingDto } from '@types';
import ROLES from '@constants/roles';

export interface IChangedParam extends Pick<SettingDto, 'clientAppCode' | 'key' | 'value' | 'userRole'> {
    type: SETTINGS_TYPES;
}

export const showNotify = (message: React.ReactNode, isError?: boolean) => {
    const config: ArgsProps = {
        duration: 5,
        message,
    };

    if (isError) {
        customNotifications.error(config);
    } else {
        customNotifications.success(config);
    }

};

type UpdateAndAddSettings = {
    updateSettings: SettingDto[];
    addSettingsArr: SettingDto[];
};

export async function createOrUpdateKey(changedParams: IChangedParam[]) {
    const { updateSettings, addSettingsArr } = changedParams.reduce<UpdateAndAddSettings>((result, { type, ...params }) => {
        const key = type === SETTINGS_TYPES.EDIT ? 'updateSettings' : 'addSettingsArr';

        if (params.key === 'referralTokenLifetime') {
            delete params.clientAppCode;
            params.key = 'token_lifetime';
            params.userRole = ROLES.REFERAL_LINK;
        }

        return {
            ...result,
            [key]: [...result[key], params],
        };
    }, { updateSettings: [], addSettingsArr: [] });

    updateSettings.length && (await updateSettingsList(updateSettings));
    addSettingsArr.length && (await addSettings(addSettingsArr));
}
