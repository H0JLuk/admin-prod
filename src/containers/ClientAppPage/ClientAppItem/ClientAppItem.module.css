.cardWrapper {
    position: relative;
    display: flex;
    margin-top: 15px;
    font-family: SBSansText-medium;
    background-color: #ffffff;
    border-radius: 5px;
    cursor: pointer;
    letter-spacing: -0.3px;
}

.cardWrapper:hover {
    box-shadow: 0 0 2.5px rgba(0, 0, 0, 0.05);
    background-color: #F8F8F8;
}

.cardInfo {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 20px;
    width: 100%;
}

.cardName {
    color: #171725;
    font-size: 16px;
    padding-right: 25px;
    flex-basis: 45%;
}

.cardCode {
    display: flex;
    flex-direction: column;
    flex-basis: 45%;
}

.cardCodeLabel {
    color: #657182;
    font-size: 16px;
}

.cardCodeValue {
    color: #171725;
    font-size: 16px;
}

.cardAction {
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
    right: 3%;
}

.cardMenu span[role='img'] {
    font-size: 26px;
    color: #9ba3ae;
}