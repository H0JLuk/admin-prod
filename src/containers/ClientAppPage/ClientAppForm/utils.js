import { notification } from 'antd';
import { addSettings, updateSettingsList } from '../../../api/services/settingsService';
import { designKeysForCheck, SETTINGS_TYPES } from './ClientAppFormConstants';

export const showNotify = (message, isError) => {
    const config = {
        duration: 5,
        placement: 'bottomRight',
    };

    if (isError) {
        notification.error({
            ...config,
            message,
        });
    } else {
        notification.success({
            ...config,
            message,
        });
    }

};

export async function createOrUpdateKey (changedParams) {
    const updateSettings = [];
    const addSettingsArr = [];

    changedParams.forEach(({ type, ...params }) => {
        type === SETTINGS_TYPES.EDIT
            ? updateSettings.push(params)
            : addSettingsArr.push(params);
    });

    updateSettings.length && (await updateSettingsList(updateSettings));
    addSettingsArr.length && (await addSettings(addSettingsArr));
}

export const checkExistDesignSettings = (settings) => {
    return designKeysForCheck.some(key => !Object.prototype.hasOwnProperty.call(settings, key));
};

export const getImage = (imageName) => {
    return `${process.env.PUBLIC_URL}/images/clientAppDesign/${imageName}.png`;
};


