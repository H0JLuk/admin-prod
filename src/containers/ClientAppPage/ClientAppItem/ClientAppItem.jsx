import React from 'react';
import { MenuOutlined } from '@ant-design/icons';
import ClientAppMenu from '../ClientAppMenu/ClientAppMenu';
import { getLinkForPromoCampaignPage } from '../../../utils/appNavigation';

import style from './ClientAppItem.module.css';
import { saveAppCode } from '../../../api/services/sessionService';

const LABEL = {
    CODE: 'Код',
};

const sortableIcon = <MenuOutlined className={ style.sortableIcon } />;

const ClientAppItem = ({ item, isSortable, matchUrl, forceUpdate, history, tooltipIsVisible }) => {
    const { code, displayName } = item ?? {};

    const onCardClick = () => {
        saveAppCode(code);
        history.push(getLinkForPromoCampaignPage());
    };

    return (
        <div className={ style.cardWrapper }>
            <div className={ style.cardInfo } onClick={ onCardClick }>
                <div className={ style.cardName }>
                    <h4>{ displayName }</h4>
                </div>
                <div className={ style.cardCode }>
                    <span className={ style.cardCodeLabel }>{ LABEL.CODE }</span>
                    <span className={ style.cardCodeValue }>{ code }</span>
                </div>
                {/* Empty div needed for correct `justify-content: space-between;` for `cardInfo` block */}
                <div />
            </div>
            { tooltipIsVisible && (
                <div className={ style.cardAction }>
                    <div className={ style.cardMenu }>
                        { isSortable ? sortableIcon : (
                            <ClientAppMenu
                                clientAppItem={ item }
                                matchUrl={ matchUrl }
                                forceUpdate={ forceUpdate }
                                history={ history }
                            />
                        )}
                    </div>
                </div>
            ) }
        </div>
    );
};

export default ClientAppItem;
