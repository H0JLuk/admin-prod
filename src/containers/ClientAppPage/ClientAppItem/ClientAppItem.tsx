import React from 'react';
import { MenuOutlined } from '@ant-design/icons';
import ClientAppMenu from '../ClientAppMenu/ClientAppMenu';
import { getLinkForPromoCampaignPage } from '../../../utils/appNavigation';
import { saveAppCode } from '../../../api/services/sessionService';
import { History } from 'history';
import { ClientAppDto } from '@types';

import style from './ClientAppItem.module.css';

const LABEL = {
    CODE: 'Код',
};

const sortableIcon = <MenuOutlined className={style.sortableIcon} />;

type ClientAppItemProps = {
    item: ClientAppDto;
    isSortable: boolean;
    matchPath: string;
    forceUpdate: () => void;
    history: History;
    tooltipIsVisible: boolean;
};

const ClientAppItem: React.FC<ClientAppItemProps> = ({
    item,
    isSortable,
    matchPath,
    forceUpdate,
    history,
    tooltipIsVisible,
}) => {
    const { code, displayName } = item ?? {};

    const onCardClick = () => {
        saveAppCode(code);
        history.push(getLinkForPromoCampaignPage());
    };

    return (
        <div className={style.cardWrapper}>
            <div className={style.cardInfo} onClick={onCardClick}>
                <div className={style.cardName}>
                    <h4>{displayName}</h4>
                </div>
                <div className={style.cardCode}>
                    <span className={style.cardCodeLabel}>{LABEL.CODE}</span>
                    <span className={style.cardCodeValue}>{code}</span>
                </div>
                {/* Empty div needed for correct `justify-content: space-between;` for `cardInfo` block */}
                <div />
            </div>
            {tooltipIsVisible && (
                <div className={style.cardAction}>
                    <div className={style.cardMenu}>
                        {isSortable ? sortableIcon : (
                            <ClientAppMenu
                                clientAppItem={item}
                                matchPath={matchPath}
                                forceUpdate={forceUpdate}
                                history={history}
                            />
                        )}
                    </div>
                </div>
            )}
        </div>
    );
};

export default ClientAppItem;
