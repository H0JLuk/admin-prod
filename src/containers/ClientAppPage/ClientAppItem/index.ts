import ClientAppItem from './ClientAppItem';

export default ClientAppItem;
