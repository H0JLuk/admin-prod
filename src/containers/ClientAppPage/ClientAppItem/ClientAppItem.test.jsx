import React from 'react';
import { shallow } from 'enzyme';
import ClientAppItem from './ClientAppItem';

import * as appNavigation from '../../../utils/appNavigation';
import * as sessionService from '../../../api/services/sessionService';

sessionService.saveAppCode = jest.fn();

describe('<ClientAppItem /> test', () => {
    const props = {
        item: { code: 'code', displayName: 'displayName' },
        isSortable: false,
        matchUrl: 'matchUrl',
        forceUpdate: jest.fn(),
        history: jest.fn(),
        tooltipIsVisible: true,
    };

    it('should match snapshot and render text props ', () => {
        const ClientAppItemComponent = shallow(<ClientAppItem { ...props } />);

        expect(
            ClientAppItemComponent.find('h4')
                .first()
                .text()
        ).toEqual(props.item.displayName);
        expect(
            ClientAppItemComponent.find('.cardCodeValue')
                .first()
                .text()
        ).toEqual(props.item.code);

        expect(ClientAppItemComponent.html()).toMatchSnapshot();
    });

    it('should call history push', () => {
        const updateProps = { ...props, history: { push: jest.fn() } };
        const ClientAppItemComponent = shallow(
            <ClientAppItem { ...updateProps } />
        );

        appNavigation.getLinkForPromoCampaignPage = jest.fn(() => 'testRole');

        ClientAppItemComponent.find('.cardInfo')
            .first()
            .simulate('click');

        const historyPushFunc = updateProps.history.push;
        expect(historyPushFunc).toBeCalledTimes(1);
        expect(historyPushFunc).toBeCalledWith('testRole');

        expect(appNavigation.getLinkForPromoCampaignPage).toBeCalledTimes(1);

        expect(sessionService.saveAppCode).toBeCalledTimes(1);
        expect(sessionService.saveAppCode).toBeCalledWith(props.item.code);
    });

    it('should turn to sortable mode', () => {
        const updateProps = { ...props, isSortable: true };
        const ClientAppItemComponent = shallow(
            <ClientAppItem { ...updateProps } />
        );

        expect(ClientAppItemComponent.exists('.sortableIcon')).toBe(true);
        expect(ClientAppItemComponent.html()).toMatchSnapshot();
    });

    it('should hide tooltip', () => {
        const updateProps = { ...props, tooltipIsVisible: false };
        const ClientAppItemComponent = shallow(
            <ClientAppItem { ...updateProps } />
        );

        expect(ClientAppItemComponent.exists('.cardAction')).toBe(false);
        expect(ClientAppItemComponent.html()).toMatchSnapshot();
    });
});
