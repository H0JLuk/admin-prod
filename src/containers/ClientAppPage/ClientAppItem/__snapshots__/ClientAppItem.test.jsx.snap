// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClientAppItem /> test should hide tooltip 1`] = `"<div class=\\"cardWrapper\\"><div class=\\"cardInfo\\"><div class=\\"cardName\\"><h4>displayName</h4></div><div class=\\"cardCode\\"><span class=\\"cardCodeLabel\\">Код</span><span class=\\"cardCodeValue\\">code</span></div><div></div></div></div>"`;

exports[`<ClientAppItem /> test should match snapshot and render text props  1`] = `"<div class=\\"cardWrapper\\"><div class=\\"cardInfo\\"><div class=\\"cardName\\"><h4>displayName</h4></div><div class=\\"cardCode\\"><span class=\\"cardCodeLabel\\">Код</span><span class=\\"cardCodeValue\\">code</span></div><div></div></div><div class=\\"cardAction\\"><div class=\\"cardMenu\\"><div class=\\"ant-dropdown-trigger menuButton dropdown\\"><span role=\\"img\\" aria-label=\\"ellipsis\\" class=\\"anticon anticon-ellipsis\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" class=\\"\\" data-icon=\\"ellipsis\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z\\"></path></svg></span></div></div></div></div>"`;

exports[`<ClientAppItem /> test should turn to sortable mode 1`] = `"<div class=\\"cardWrapper\\"><div class=\\"cardInfo\\"><div class=\\"cardName\\"><h4>displayName</h4></div><div class=\\"cardCode\\"><span class=\\"cardCodeLabel\\">Код</span><span class=\\"cardCodeValue\\">code</span></div><div></div></div><div class=\\"cardAction\\"><div class=\\"cardMenu\\"><span role=\\"img\\" aria-label=\\"menu\\" class=\\"anticon anticon-menu sortableIcon\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" class=\\"\\" data-icon=\\"menu\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z\\"></path></svg></span></div></div></div>"`;
